RIOT-OS/RIOT
# makefiles/vars.inc.mk: move definition and export of CPU_FAM and CPU_ARCH to vars.inc.mk

### Contribution description

This PR adds definitons for `CPU_FAM` and `CPU_ARCH` in  `makefiles/vars.inc.mk` and removes the exports in `cpu` `boards` so there are only exported once.

This is a stepping stone for moving `CPU_FAM` and `CPU_ARCH` to `Makefiles.features` and remove the export.

### Testing procedure

The variables are still exported, so no change in behavior. The commits should be individually reviewed.

You can also see that nothing has changed for an applicaiton:

This pr:
```
for board in $(make info-boards); do BOARD=${board} make --no-print-directory -C examples/hello-world/ info-debug-variable-BOARD info-debug-variable-CPU_FAM info-debug-variable-CPU_ARCH; done > pr.txt
```

In master:
```
for board in $(make info-boards); do BOARD=${board} make --no-print-directory -C examples/hello-world/ info-debug-variable-BOARD info-debug-variable-CPU_FAM info-debug-variable-CPU_ARCH; done > master.txt
```

And check the is no diff:

```
diff master.txt pr.txt
#no diff 
```

### Issues/PRs references

Related to #9913.
